action.yml

```diff
@@ -91,6 +94,14 @@ runs:
             --token ${{ steps.get-registration.outputs.registration_token }} --name $(hostname) --work /home/ghrunner \
             --labels ${{ steps.get-registration.outputs.runner_scope }} --ephemeral
 
+          # Running custom script if provided
+          if [[ -n "${{ inputs.custom_script }}" ]]; then
```

Comment:
I think we'll run into problems by parking a whole script in the conditional args. Perhaps it would be better to write to the custom_script first, conditionally make it executable, and run it only if the file size is greater than 0. I have similar concerns about the custom_script making it from environment variable to file via echo.

Reply:
Perhaps:

```sh
cat >  /tmp/custom_script.sh <<'CUSTOM_SCRIPT_EOF'
${{ inputs.custom_script }}
CUSTOM_SCRIPT_EOF
```

This will make the custom_script free to use subshells, backticks, and variables without the parent shell interpolating them.

Reply:
It may also be better use use `mktemp`

Reply:
> I think we'll run into problems by parking a whole script in the conditional args

Can you elaborate more on this? IMO `if [[ -n "${{ inputs.custom_script }}" ]]` will be a better way to check if we have specified the custom script or not. Creating an empty file in case no custom script is provided is unnecessary.

Reply:
A multiline script would break the if statement. Any subcommands, variables, or redirection included in the custom_script would be evaluated during the conditional, before the custom_script is run.

```sh
$ if [[ -n "echo "$(echo hello `echo world`>/tmp/t)"; " ]]; then echo foo; fi
foo
$ ls -l /tmp/t
-rw-r--r--  1 mjohansson  wheel  12 Sep 27 10:04 /tmp/t
```

Reply:
```
# Running custom bash script if provided
exec 3<<'CUSTOM_SCRIPT_EOF'
${{ inputs.custom_script }}
CUSTOM_SCRIPT_EOF

bash <&3
exec 3<&-
```

Reply:
@displague How about this?
```
script='${{ inputs.custom_script }}'
  if [[ -n "$script" ]]; then
    CUSTOM_SCRIPT=$(mktemp $HOME/custom_script.XXXXXX.sh)
     printf '%s\n' "$script" >> "$CUSTOM_SCRIPT"
     chmod 0700 $CUSTOM_SCRIPT
     echo "Running custom user script"
     $CUSTOM_SCRIPT
  fi
```
This solves the problem of not accidentally running the script when the condition is checked while also making sure that original script is interpreted and used as it?